source/server/active_udp_listener.h

```diff
@@ -103,6 +103,12 @@ class ActiveRawUdpListener : public ActiveUdpListenerBase,
     read_filter_.reset();
     udp_listener_.reset();
   }
+  void updateListenerConfig(Network::ListenerConfig&) override {
+    ENVOY_BUG(false, "In-place filter chain update observed on raw UDP listener.");
```

Comment:
Can this happen (if the user intentionally does it)? Or should this have been caught at some other layer?

Reply:
If raw UDP listener uses L4 filters, this could happen. I don't think we caught this case at another layer since this PR unblocked updating a UDP listener config.

Reply:
Things that the user can do shouldn't be an `ENVOY_BUG`. 

Is this called in a path where an exception can be thrown to signal failure of the config update? If not, can we catch this from somewhere that an exception can be thrown? At a minimum, change this to an error-level log message.

Reply:
Do we want to crash Envoy during in-place config update if the config is bad?

Reply:
No, definitely not. If throwing an exception here crashes instead of reporting a config load failure, then this is too late in the process, and we'll need to catch it earlier.

Reply:
In fact, ActiveUdpListener is connectionless, so it shouldn't have filter chain configured to begin with. So I don't think this code is reachable.

Reply:
If the code is not reachable than make it NOT_REACHED with a comment? Otherwise I agree this shouldn't be an ENVOY_BUG and it should either:
1) Optimally somehow fail the config update (throw exception if in that path, but that might not be possible)
2) Switch to error or warn that the config update is being ignored. 
Same below.

Having read the rest of the PR, I wonder if you could error out here on the path that we choose to do an in-place update when we get the LDS update if it is UDP and not QUIC? That's if we can actually get there. Then these can definitely be NOT_REACHED with appropriate comments?

Reply:
Good point! After thinking it more, I realize we should always reject non-QUIC UDP listener config with filter chain regardless whether it is initial start or config update.